### ISSUE TITLE
Cannot connect to syslog on virtual interface

### ISSUE THREAD

**Issue by a-validnerd** [balabit/syslog-ng-docker#27]
Hello,

I have a a normal linux interface with an IP of 2.2.2.120. I then add a virtual interface with an IP of 2.2.2.121.
night@ubuntu-westeros-com:~$ ifconfig
docker0   Link encap:Ethernet  HWaddr 02:42:0c:d9:69:74
          inet addr:172.17.0.1  Bcast:0.0.0.0  Mask:255.255.0.0

ens36     Link encap:Ethernet  HWaddr 00:50:56:86:8c:6d
          inet addr:2.2.2.120  Bcast:2.2.2.255  Mask:255.255.255.0

ens36:1   Link encap:Ethernet  HWaddr 00:50:56:86:8c:6d
          inet addr:2.2.2.121  Bcast:2.2.2.255  Mask:255.255.255.0
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1

I then start the balabit/syslog-ng container:
# sudo docker run -it -p 2.2.2.121:514:514 -p 2.2.2.121:601:601 --name syslog121 balabit/syslog-ng -edv --no-caps

The container starts successfully and I can see on the Linux host that it's listening  (although only on TCP):
$ sudo netstat -pano | grep 2.2.2.121
tcp        0      0 2.2.2.121:601           0.0.0.0:*               LISTEN      26106/docker-proxy off (0.00/0/0)
tcp        0      0 2.2.2.121:514           0.0.0.0:*               LISTEN      26119/docker-proxy off (0.00/0/0)

When I try to telnet from a different host, I get a RST (I run a ping before so you can see I have reachability)
$ sudo tcpdump -nni any host 2.2.2.31

03:50:48.478289 IP 2.2.2.31 > 2.2.2.121: ICMP echo request, id 38161, seq 1, length 64
03:50:48.478575 IP 2.2.2.121 > 2.2.2.31: ICMP echo reply, id 38161, seq 1, length 64

03:50:48.484432 IP 2.2.2.31.54908 > 2.2.2.121.514: Flags [S], seq 2389207154, win 14600, options [mss 1460,sackOK,TS val 1456495683 ecr 0,nop,wscale 7], length 0
03:50:48.484644 IP 2.2.2.31.54908 > 172.17.0.2.514: Flags [S], seq 2389207154, win 14600, options [mss 1460,sackOK,TS val 1456495683 ecr 0,nop,wscale 7], length 0
03:50:48.484656 IP 2.2.2.31.54908 > 172.17.0.2.514: Flags [S], seq 2389207154, win 14600, options [mss 1460,sackOK,TS val 1456495683 ecr 0,nop,wscale 7], length 0
03:50:48.484727 IP 172.17.0.2.514 > 2.2.2.31.54908: Flags [R.], seq 0, ack 2389207155, win 0, length 0
03:50:48.484727 IP 172.17.0.2.514 > 2.2.2.31.54908: Flags [R.], seq 0, ack 1, win 0, length 0
03:50:48.484757 IP 2.2.2.121.514 > 2.2.2.31.54908: Flags [R.], seq 0, ack 2389207155, win 0, length 0

The RST is sent from 172.17.0.2 (which is the IP assigned to the container)
$ sudo docker inspect --format '{{ .NetworkSettings.IPAddress }}' syslog121
172.17.0.2

I don't get any other errors.  This image works: https://hub.docker.com/r/bobrik/syslog-ng/.

Cheers,
V

**Comment by Kokan** [comment]
Hello,

Could you share your syslog-ng.conf ?
Do you start the bobrik/syslog-ng docker image with the same parameters as the syslog121  container ?

> The container starts successfully and I can see on the Linux host that it's listening (although only on TCP):

Because -p ip:port:port uses tcp by default, ip:port:port/udp solves this :)

**Comment by a-validnerd** [comment]
Hello,

1. Thanks for the UDP tip :).
2. It seems that the default syslog-ng.conf file in the container only listens for local logs:
https://pastebin.com/quqVsuda

I tried to comment the "source s_src" directive and added these lines instead (in my mind these are supposed to accept anything coming to TCP 514 and write to /var/log/messages).
>source s_net { tcp(port(514)); };
>destination d_net { file("/var/log/messages"); };
>log { source(s_net); destination(d_net); };

I had to edit the file on the Linux host, then copy it back to the container and restarted the daemon (service syslog-ng restart)
But I still get the same RST:
03:16:14.934506 IP 2.2.2.31.43840 > 2.2.2.121.514: Flags [S], seq 1041331219, win 14600, options [mss 1460,sackOK,TS val 1540822727 ecr 0,nop,wscale 7], length 0
03:16:14.934642 IP 2.2.2.31.43840 > 172.17.0.2.514: Flags [S], seq 1041331219, win 14600, options [mss 1460,sackOK,TS val 1540822727 ecr 0,nop,wscale 7], length 0
03:16:14.934653 IP 2.2.2.31.43840 > 172.17.0.2.514: Flags [S], seq 1041331219, win 14600, options [mss 1460,sackOK,TS val 1540822727 ecr 0,nop,wscale 7], length 0
03:16:14.935245 IP 172.17.0.2.514 > 2.2.2.31.43840: Flags [R.], seq 0, ack 1041331220, win 0, length 0
03:16:14.935245 IP 172.17.0.2.514 > 2.2.2.31.43840: Flags [R.], seq 0, ack 1, win 0, length 0
03:16:14.935311 IP 2.2.2.121.514 > 2.2.2.31.43840: Flags [R.], seq 0, ack 1041331220, win 0, length 0

Regards,
V
